CDS/js/Datastore.js

```diff
@@ -114,24 +116,100 @@ function get(conn,type, key, callback, userdata) { /* callback as function(err,
                         console.error(err);
                         reject(err);
                     }
+                    var rtn;
                     if (!row) {
-                        fulfill({value: "(value not found)",type: type,key: key});
+                        rtn = {value: "(value not found)",type: type,key: key};
                     } else {
-                        if(callback){
-                            fulfill(callback(row.value,type,key,userdata));
-                        } else {
-                            fulfill({value: row.value,type: type,key: key});
-                        }
+                        rtn = {value: row.value,type: type,key: key}
+                    }
```

Comment:
minor - this block can be simplified a little by:

```js
const rtn = {
  type: type,
  key: key
}

rtn.value = row ? row.value : '(value not found)';
```

(it also removes a nasty `var` 😝 )